server.go

```diff
@@ -103,21 +103,22 @@ type Response struct {
 	ErrType responseErr
 }
 
-// Permissions maps "service.method" of a peer to boolean permission
-type Permissions map[peer.ID]map[string]bool
-
-// AuthorizeWithMap returns an authrorization function that follows the authorization strategy as described in the
+// AuthorizeWithMap returns an authrorization function that follows the strategy as described in the
 // given map(maps "service.method" of a peer to boolean permission)
-func AuthorizeWithMap(p Permissions) func(peer.ID, string, string) bool {
-	if p != nil {
-		return func(pid peer.ID, svc string, method string) bool {
-			return p[pid][svc+"."+method]
+func AuthorizeWithMap(p map[peer.ID]map[string]bool) func(peer.ID, string, string) bool {
+	return func(pid peer.ID, svc string, method string) bool {
+		// If map is nil, no method would be allowed
+		if p == nil {
```

Comment:
Based on previous comments, nil maps behave as empty maps, so this is not necessary here I think.

Reply:
I was under assumption that the main map would have to be not nil, but weirdly not. You are right.